api/test/wfl/integration/sinks/datarepo_sink_test.clj

```diff
@@ -119,13 +119,19 @@
     (assertion #(util/poll task (or interval 1) (or times 5)))))
 
 (deftest test-update-datarepo-sink
-  (let [description (resources/read-resource "primitive.edn")
-        workflow    {:uuid (UUID/randomUUID) :outputs outputs}
-        upstream    (make-queue-from-list [[description workflow]])
-        sink        (create-and-load-datarepo-sink)]
+  (let [description      (resources/read-resource "primitive.edn")
+        workflow         {:uuid (UUID/randomUUID) :outputs outputs}
+        upstream         (make-queue-from-list [[description workflow]])
+        failing-workflow {:uuid (UUID/randomUUID) :outputs {:outbool   true
+                                                            :outfile   "gs://not-a-real-bucket/external-reprocessing/exome/develop/not-a-real.unmapped.bam"
+                                                            :outfloat  (* 4 (Math/atan 1))
+                                                            :outint    27
+                                                            :outstring "Hello, World!"}}
+        failing-upstream (make-queue-from-list [[description failing-workflow]])
+        sink             (create-and-load-datarepo-sink)]
     (sink/update-sink! upstream sink)
     (is (stage/done? upstream))
-    (eventually (throws? UserException) #(sink/update-sink! upstream sink)
+    (is (or (sink/update-sink! upstream sink) true) "subsequent updates do nothing")
```

Comment:
Do we want something that tests that eventually the good job succeeds?  That would be a nice addition here. (But maybe not a release blocker.)

Reply:
A few more thoughts.

I think that "subsequent updates do nothing" does not adequately describe the behavior here, or what a test failure would mean.

Wouldn't `(or (sink/update-sink! upstream sink) true)` always return true?

Reply:
(And, now I see that you didn't add this test initially.  Sorry to pull you into it!)

Reply:
Apologies for the "thinking out loud" in this thread.

I think initially, the purpose of this check following the `(eventually <updating this sink yields a UserException>)` check is to confirm that once a job is identified as having failed, we won't continue to throw exceptions for it.  So in its current location it's not giving us anything useful.

Reply:
I think you are correct. I will move it back down to under the error check and remove the or and see what happens.

Reply:
The `or` might be there if `sink/update-sink!` is executed for side effects and returns `nil` in normal operations.  Again with the thinking out loud here :)

Reply:
Sorry, yea. I realized that after my comment. I am leaving it. The real test for that line is whether it throws another exception and not the return of the function. I am also adding a check for the status of the job. Should have it up in a few minutes.